godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -457,6 +457,51 @@ macro_rules! impl_vector_fns {
     }
 }
 
+pub(super) fn snap_one(mut value: i32, step: i32) -> i32 {
+    assert!(
+        value != i32::MIN || step != -1,
+        "snapped() called on vector component i32::MIN with step component -1"
+    );
```

Comment:
Is only this a problem? `step == -1` would work?

Reply:
No, I just realized how many overflow errors this function has... I have no idea how to handle all these, maybe you could give some guidance what the best way forward would be?

Reply:
Limit the input range to the common usage (and panic for others) would be a start. Upon user request, we can then expand in the future.